submissions/ses3332002/tiny-js-world/index.js

```diff
@@ -0,0 +1,106 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+const dog = {
+  species: "dog",
+  name: "Toby",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Woof-woof!",
+  friends: [
+    "Puss"
+  ]
+};
+
+const cat = {
+  species: "cat",
+  name: "Puss",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Meow!",
+  friends: [
+    "Toby",
+    "Halley"
+  ]
+};
+
+const man = {
+  species: "human",
+  name: "Jack",
+  gender: "male",
+  legs: 2,
+  hands: 2,
+  saying: "Hey baby!",
+  friends: [
+    "Jennie",
+    "Halley"
+  ]
+};
+
+const woman = {
+  species: "human",
+  name: "Jennie",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  saying: "Hi darling!",
+  friends: [
+    "Jack",
+    "Toby"
+  ]
+};
+
+const catWoman = {
+  species: "super-hero",
+  name: "Halley",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  saying: cat.saying,
+  friends: [
+    "Puss"
+  ]
+};
+
+const inhabitants = [dog, cat, man, woman, catWoman];
+const propsToPrint = ["species", "name", "gender", "legs", "hands", "saying", "friends"];
+
+inhabitants.forEach((item) => {
+  let description = "";
+  propsToPrint.forEach((prop) => {
+    if (Array.isArray(item[prop])) {
+      description = description + item[prop].join(', ');
+    } else {
+      description = description + item[prop] + "; ";
+    };
+  });
+  print(description);
```

Comment:
`.forEach` assumes side effects (e.g. reference to variables defined beyond the forEach callback). Side effects should be avoided whenever possible.

String concatenation is an error prone approach (like here, freinds list will never be terminated with `';'` if the property appears in the middle of the list).

Consider the following simplified fragments to build a cleaner code.

`someString = props.map(propName => item[propName] /* and other magic for array props*/).join("some delimiter");`

Consider mastering `Array` methods. They are powerful and help creating cleaner code.

Reply:
I haven't seen your review yet, but I was in a good mood :) and improved the code. 
Now beautiful output does not depend on where the array is located and whether it exists at all.